### PR TITLE
Fix Issue 19092 - __delete doesn't work with immutable

### PR DESCRIPTION
**Pull request by wilzbach**

### CODE REVIEW

File: src/core/memory.d [hunk 1]
```diff
@@ -1024,6 +1025,26 @@ void __delete(T)(ref T x) @system
     }
 }
 
+/// ditto
+void __delete(T)(ref const(T)[] x) @system
+if (is(const(T) : T))
+{
+    import core.internal.traits : Unqual;
+    auto x2 = cast(Unqual!T[]) x;
+    __delete(x2);
```
Comment (wilzbach) [inline src/core/memory.d]: We could also do `GC.free(cast(void*) x);` directly here.

File: src/core/memory.d [hunk 2]
```diff
@@ -1024,6 +1025,26 @@ void __delete(T)(ref T x) @system
     }
 }
 
+/// ditto
+void __delete(T)(ref const(T)[] x) @system
+if (is(const(T) : T))
+{
+    import core.internal.traits : Unqual;
+    auto x2 = cast(Unqual!T[]) x;
```
Comment (jacob-carlborg) [inline src/core/memory.d]: Why not doing this in the mutable overload?

Comment (wilzbach) [reply src/core/memory.d]: Fair point!

**Comment by jacob-carlborg** [comment]
Although the issue doesn't mention this, but it doesn't work for const/immutable classes, interfaces or pointers as well.

**Comment by wilzbach** [comment]
> Although the issue doesn't mention this, but it doesn't work for const/immutable classes, interfaces or pointers as well.

Yes, but `delete` didn't support this either: https://run.dlang.io/is/gKbcrq

**Comment by wilzbach** [comment]
Failures are unrelated :/

**Pull request merged**